url433bec1e56_sarobe_DifficultyInvestigation_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/fr/inria/optimization/cmaes/CMAEvolutionStrategy.java

```diff
@@ -0,0 +1,2704 @@
+    
+          double scale = 0.0;
+          double h = 0.0;
+          for (int k = 0; k < i; k++) {
+             scale = scale + Math.abs(d[k]);
+          }
+          if (scale == 0.0) {
+             e[i] = d[i-1];
+             for (int j = 0; j < i; j++) {
+                d[j] = V[i-1][j];
+                V[i][j] = 0.0;
+                V[j][i] = 0.0;
+             }
+          } else {
+    
+             // Generate Householder vector.
+    
+             for (int k = 0; k < i; k++) {
+                d[k] /= scale;
+                h += d[k] * d[k];
+             }
+             double f = d[i-1];
+             double g = Math.sqrt(h);
+             if (f > 0) {
+                g = -g;
+             }
+             e[i] = scale * g;
+             h = h - f * g;
+             d[i-1] = f - g;
+             for (int j = 0; j < i; j++) {
+                e[j] = 0.0;
+             }
+    
+             // Apply similarity transformation to remaining columns.
+    
+             for (int j = 0; j < i; j++) {
+                f = d[j];
+                V[j][i] = f;
+                g = e[j] + V[j][j] * f;
+                for (int k = j+1; k <= i-1; k++) {
+                   g += V[k][j] * d[k];
+                   e[k] += V[k][j] * f;
+                }
+                e[j] = g;
+             }
+             f = 0.0;
+             for (int j = 0; j < i; j++) {
+                e[j] /= h;
+                f += e[j] * d[j];
+             }
+             double hh = f / (h + h);
+             for (int j = 0; j < i; j++) {
+                e[j] -= hh * d[j];
+             }
+             for (int j = 0; j < i; j++) {
+                f = d[j];
+                g = e[j];
+                for (int k = j; k <= i-1; k++) {
+                   V[k][j] -= (f * e[k] + g * d[k]);
+                }
+                d[j] = V[i-1][j];
+                V[i][j] = 0.0;
+             }
+          }
+          d[i] = h;
+       }
+    
+       // Accumulate transformations.
+    
+       for (int i = 0; i < n-1; i++) {
+          V[n-1][i] = V[i][i];
+          V[i][i] = 1.0;
+          double h = d[i+1];
+          if (h != 0.0) {
+             for (int k = 0; k <= i; k++) {
+                d[k] = V[k][i+1] / h;
+             }
+             for (int j = 0; j <= i; j++) {
+                double g = 0.0;
+                for (int k = 0; k <= i; k++) {
+                   g += V[k][i+1] * V[k][j];
+                }
+                for (int k = 0; k <= i; k++) {
+                   V[k][j] -= g * d[k];
+                }
+             }
+          }
+          for (int k = 0; k <= i; k++) {
+             V[k][i+1] = 0.0;
+          }
+       }
+       for (int j = 0; j < n; j++) {
+          d[j] = V[n-1][j];
+          V[n-1][j] = 0.0;
+       }
+       V[n-1][n-1] = 1.0;
+       e[0] = 0.0;
+    } 
+
+    // Symmetric tridiagonal QL algorithm, taken from JAMA package.
+    
+    private void tql2 (int n, double d[], double e[], double V[][]) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1378-1396 into a separate method.